TELCODOCS-382 - release note known issues for OCP 4.9.6 z-stream RAN GA release (#38300)
openshift/openshift-docs: release_notes/ocp-4-9-release-notes.adoc

* There is an instability issue with the inbox iavf driver that is evident when a DPDK based workload is deployed in {product-title} version 4.8.5. The issue is also apparent when a DPDK workload is deployed on a host running RHEL for Real Time 8. The issue occurs in hosts with Intel XXV710 NICs installed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2000180[BZ#2000180])

* An error occurs during SNO cluster provisioning where Bootkube tries to use `oc` towards the end of the cluster bootstrap process. The Kube API receives a shutdown request and this causes the cluster bootstrap process to fail. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2010665[BZ#2010665])

* A clock jump error occurs in the `linuxptp` subsystem that is deployed by the PTP Operator. The reported error is: "clock jumped backward or running slower than expected!". The error is encountered in a host with an Intel E810 NIC installed in a {product-title} version 4.8 cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2013478[BZ#2013478])

> * Deploying an {product-title} version 4.9 SNO cluster after a successful 4.8 deployment on the same host fails due to a modified boot table entry. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2011306[BZ#2011306])

Reviewer: I have run into the same issue deploying back-to-back 4.9 cluster installs.

Author: Thanks @imiller0 - I've updated the known issue slightly to reflect this.

* The `oc adm release extract --tools` command fails when an image that is hosted in the local registry is specified. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1823143[BZ#1823143])

* In a Single Node OpenShift (SNO) cluster that uses real-time kernel version 8.4, the time taken to create a pod is two and a half times slower than with the standard kernel version 8.4. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975356[BZ#1975356])

* In a SNO cluster deployed with 150 or more pods, pod recovery times are slow after a node reboot. The slow recovery is due to the following error: "timed out waiting for OVS port binding". When the issue occurs there are usually mutiple failures. The pods eventually recover. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1986216[BZ#1986216])

* An error occurs during SNO cluster provisioning where Bootkube tries to use `oc` towards the end of the cluster bootstrap process. The Kube API receives a shutdown request and this causes the cluster bootstrap process to fail. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2010665[BZ#2010665])

> * Deploying an {product-title} version 4.9 SNO cluster after a successful 4.8 or 4.9 deployment on the same host fails due to a modified boot table entry. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2011306[BZ#2011306])

Reviewer 2: I think "or 4.9" can be removed here. The unexpected boot entry is only created during 4.8 install as qe observed.

Author: FYI @imiller0

+
Optionally, you can update the image registry settings by link:https://docs.openshift.com/container-platform/4.9/openshift_images/image-configuration.html#images-configuration-shortname_image-configuration[adding registries that allow image short names]. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2011293[*BZ#2011293*])

* The `oc adm release extract --tools` command fails when an image that is hosted in the local registry is specified. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1823143[BZ#1823143])

* On an {product-title} single node configuration, pod creation times are over two times slower when using the real-time kernel (`kernel-rt`) than when using the non-real time kernel. When using `kernel-rt`, the slower pod creation times affect the maximum number of supported pods because recovery time is impacted after a node reboots.
+

> As a workaround, when you use `kernel-rt`, you can improve the recovery time by booting with the `rcupdate.rcu_normal_after_boot=0` kernel argument. This requires a real-time kernel version `kernel-rt-4.18.0-305.16.1.rt7.88.el8_4` or later. This known issue applies to {product-title} version 4.8.15 and later. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975356[BZ#1975356])

Reviewer 3: One extra thing, I think the BZ# need to be bolded, so you should apply [*BZ123456*] to make these bold.

> Following an {product-title} single node reboot, all pods are restarted which causes significant load and longer than normal pod creation times. This happens because the Container Network Interface (CNI) is not able to process the `pod add` events quickly enough. The following error message is displayed: `timed out waiting for OVS port binding`. The {product-title} single node instance eventually recovers, though more slowly than expected. This known issue applies to {product-title} version 4.8.15 and later. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1986216[BZ#1986216])

Reviewer 3: I think this should have a "*" to make it a bullet.

> * An error occurs during SNO cluster provisioning where Bootkube tries to use `oc` towards the end of the cluster bootstrap process. The Kube API receives a shutdown request and this causes the cluster bootstrap process to fail. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2010665[BZ#2010665])

Reviewer 4: ```suggestion
* An error occurs during Single Node OpenShift (SNO) cluster provisioning where bootkube tries to use `oc` towards the end of the cluster bootstrap process. The Kube API receives a shutdown request and this causes the cluster bootstrap process to fail. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2010665[BZ#2010665])
```

* Deploying an {product-title} version 4.9 SNO cluster after a successful 4.8 deployment on the same host fails due to a modified boot table entry. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2011306[BZ#2011306])

> * There is an instability issue with the inbox iavf driver that is evident when a DPDK based workload is deployed in {product-title} version 4.8.5. The issue is also apparent when a DPDK workload is deployed on a host running RHEL for Real Time 8. The issue occurs in hosts with Intel XXV710 NICs installed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2000180[BZ#2000180])

Reviewer 4: ```suggestion
* There is an instability issue with the inbox iavf driver that is evident when a DPDK-based workload is deployed in {product-title} version 4.8.5. The issue is also apparent when a DPDK workload is deployed on a host running RHEL for Real Time 8. The issue occurs in hosts with Intel XXV710 NICs installed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2000180[BZ#2000180])
```

> * A clock jump error occurs in the `linuxptp` subsystem that is deployed by the PTP Operator. The reported error message is: `clock jumped backward or running slower than expected!`. The error is encountered in a host with an Intel Columbiaville E810 NIC installed in a {product-title} version 4.8 or 4.9 cluster. The error is likely ice driver related, rather than an error in the `linuxptp` subsystem. An imminent Intel driver update and {product-title} version 4.9 support for the Columbiaville NIC will resolve this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2013478[BZ#2013478])

Reviewer 3: ```suggestion
* A clock jump error occurs in the `linuxptp` subsystem that is deployed by the PTP Operator. The reported error message is: `clock jumped backward or running slower than expected!`. The error is encountered in a host with an Intel Columbiaville E810 NIC installed in a {product-title} version 4.8 or 4.9 cluster. The error is likely ice driver related, rather than an error in the `linuxptp` subsystem. An imminent Intel ice driver update and {product-title} version 4.9 support for the Columbiaville NIC resolves this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2013478[BZ#2013478])
```

Reviewer 3: Just for consistency adding ice would be nice :) also avoid future tense if possible.

> * A clock jump error occurs in the `linuxptp` subsystem that is deployed by the PTP Operator. The reported error is: "clock jumped backward or running slower than expected!". The error is encountered in a host with an Intel Columbiaville E810 NIC installed in a {product-title} version 4.8 cluster. The error is likely ice driver related, rather than an error in the `linuxptp` subsystem. An imminent Intel firmware update and {product-title} version 4.9 support for the Columbiaville NIC is expected which will resolve this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2013478[BZ#2013478])

Reviewer 5: or 4.9

Reviewer 5: Intel driver update

+
(link:https://bugzilla.redhat.com/show_bug.cgi?id=2015829[*BZ#2015829*])

[id="ocp-4-9-ran-ga-known-issues"]
* The `oc adm release extract --tools` command fails when an image that is hosted in the local registry is specified. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1823143[*BZ#1823143*])

> * On an {product-title} single node configuration, pod creation times are over two times slower when using the real-time kernel (`kernel-rt`) than when using the non-real time kernel. When using `kernel-rt`, the slower pod creation times affect the maximum number of supported pods because recovery time is impacted after a node reboots.

Reviewer 6: ```suggestion
* On a {product-title} single node configuration, pod creation times are over two times slower when using the real-time kernel (`kernel-rt`) than when using the non-real time kernel. When using `kernel-rt`, the slower pod creation times affect the maximum number of supported pods because recovery time is impacted after a node reboots.
```

Should be "a" OpenShift Container Platform, I don't believe we refer to it as "an" OpenShift Container Platform. Double check with release notes and other docs to make sure that we are consistent with the use of "a" vs. "an". You have several instances of this, so I'm just commenting here once just make sure to update if necessary.

Author: Interesting. Are you sure about this? That would seem to go against basic grammar rules. https://www.grammarly.com/blog/indefinite-articles-a-and-an/

* There is an instability issue with the inbox iavf driver that is evident when a DPDK-based workload is deployed in {product-title} version 4.8.5. The issue is also apparent when a DPDK workload is deployed on a host running RHEL for Real Time 8. The issue occurs in hosts with Intel XXV710 NICs installed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2000180[*BZ#2000180*])

* A clock jump error occurs in the `linuxptp` subsystem that is deployed by the PTP Operator. The reported error message is: `clock jumped backward or running slower than expected!`. The error is encountered in a host with an Intel Columbiaville E810 NIC installed in a {product-title} version 4.8 or 4.9 cluster. The error is likely Intel ice driver related, rather than an error in the `linuxptp` subsystem. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2013478[*BZ#2013478*])

* Sometimes Operator installation fails during zero touch provisioning (ZTP) installation of a DU node. The `InstallPlan` API reports an error. The reported error message is: `Bundle unpacking failed. Reason: DeadlineExceeded`. The error occurs if the Operator installation job exceeds 600 seconds.
+

> To workaround this issue and re-try the Operator install, run the following `oc` commands for the failed Operator:

Reviewer 4: ```suggestion
To workaround this issue and re-try the Operator installation, run the following `oc` commands for the failed Operator:
```

+
[source,terminal]
----
$ oc -n openshift-marketplace delete catsrc <failed_operator_name>
----
+
. Delete the install plan:
+
[source,terminal]
----
$ oc -n <failed_operator_namespace> delete ip <failed_operator_install_plan>
----
+
. Delete the subscription and wait for the Operator `CatalogSource` and `Subscription` resources to be re-created by the relevant custom resource policy:
+
[source,terminal]
----
$ oc -n <failed_operator_namespace> delete sub <failed_operator_subscription>
----
+
.Expected result
+
The Operator `InstallPlan` and `ClusterServiceVersion` resources are created and the Operator is installed.
+
(link:https://bugzilla.redhat.com/show_bug.cgi?id=2021456[*BZ#2021456*])

* Sometimes when the ZTP installation process finishes provisioning a DU node, the performance profile configuration is not applied to the node and the `MachineConfigPool` resource becomes stuck in an `Updating` state.
+
To workaround this issue, perform the following procedure:
+
. Get the name of the failed DU node:
+
[source,terminal]
----
$ oc get mcp
----
+
.Example output
+
[source,terminal]
----
NAME     CONFIG                         UPDATED   UPDATING   DEGRADED

> master   rendered-master-90fe2b00c718   False     True       False

Reviewer 4: Do we need to comply with [this guideline](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#using-conscious-language) for terminal outputs too? It might be worth checking with the peer-review-squad.

Author: We can in this instance I think, I've removed the "master" reference.

compute-1         rendered-compute-1-31197fc6da09          True      False      False
----
+
. Uncordon the failed node, and wait for the `machine-config-daemon` to apply the performance profile. For example:
+
[source,terminal]
----
$ oc adm uncordon compute-compute-1-31197fc6da09
----
+
.Expected result
+
The `machine-config-daemon` applies the performance profile configuration to the node.
+

> (link:https://bugzilla.redhat.com/show_bug.cgi?id=2021151[*BZ#2021151*])

Reviewer 5: Suggested updates after 4.9 GA review today:

* A race condition exists between the SRIOV operator and the MCO which can manifest itself in different ways.  Sometimes when the ZTP installation process finishes provisioning a DU node, the performance profile configuration is not applied to the node and the `MachineConfigPool` resource becomes stuck in an `Updating` state.
+
To workaround this issue, perform the following procedure:
+
. Get the name of the failed DU node:
+
[source,terminal]
----
$ oc get mcp
----
+
.Example output
+
[source,terminal]
----
NAME              CONFIG                                   UPDATED   UPDATING   DEGRADED
control-plane-1   rendered-control-plane-1-90fe2b00c718    False     True       False
compute-1         rendered-compute-1-31197fc6da09          True      False      False
----
+
. Uncordon the failed node, and wait for the `machine-config-daemon` to apply the performance profile. For example:
+
[source,terminal]
----
$ oc adm uncordon compute-compute-1-31197fc6da09
----
+
.Expected result
+
The `machine-config-daemon` applies the performance profile configuration to the node.
+
Another manifestation of this issue is that the performance profile configuration does not get applied during DU node configuration.  To work around this issue:
+
. Apply rpm-ostree kargs manually:
+
[source,terminal]
----
$ rpm-ostree  kargs --append=skew_tick=1 --append=nohz=on --append=rcu_nocbs=2-23,26-47 --append=tuned.non_isolcpus=03000003 --append=intel_pstate=disable --append=nosoftlockup --append=tsc=nowatchdog --append=intel_iommu=on --append=iommu=pt --append=isolcpus=managed_irq,2-23,26-47 --append=systemd.cpu_affinity=0,1,24,25 --append=default_hugepagesz=1G --append=hugepagesz=1G --append=hugepages=32 --append=idle=poll --append=rcupdate.rcu_normal_after_boot=0 --append=nohz_full=2-23,26-47
----
+
. Reboot the node
+
.Expected result
+
The Performance profile is applied properly.
+
Another way this issue manifests itself is that the DU node reboot during policy configuration can take tens of minutes.  Note that no workaround is required here.  The system will eventually recover.
+
(link:https://bugzilla.redhat.com/show_bug.cgi?id=2021151[*BZ#2021151*])

Reviewer 7: /lgtm
